### DISCUSSION TITLE
Could we make it support other languages like Vala?

### DISCUSSION THREAD

**Discussion by taozuhong** [rust-lang/rust-bindgen#2651] [General]
Vala language needs a tool to generate VAPI files, could we make it support other languages like Vala?

**Comment by pvdrz** [comment]
We currently only (partially) support languages that can be compiled by clang: C, Objective-C and C++. Adding support for languages different than those would imply writing code to talk to the compiler of such language and translate its output to bindgen's IR. But at that point it might be easier to just write a different tool from scratch.

**Reply by taozuhong** [reply]
Ah, Vala compiler is a C output-based compiler, the final compiler is C compiler, so it could be supported.

**Reply by pvdrz** [reply]
even if that's the case I imagine not all language constructs map 1-to-1 to their C counterparts. That being said, you should already be able to pass whatever header file is produced by the vala compiler to bindgen.
